expression/integration_test.go

```diff
@@ -9751,3 +9751,26 @@ func (s *testIntegrationSuite) TestCharsetErr(c *C) {
 	err = tk.ExecToErr("insert into charset_test(c1) values ('aaa\xEF\xBF\xBD')")
 	c.Assert(err.Error(), Equals, "[table:1366]Incorrect string value '\\xEF\\xBF\\xBD' for column 'c1'")
 }
+
+func (s *testIntegrationSuite2) TestIssue25591(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+
+	tk.MustExec("use test;")
+	tk.MustExec("drop table if exists t1_1, t2_1;")
+	tk.MustExec("CREATE TABLE `t1_1` (`col1` double DEFAULT NULL);")
+	tk.MustExec("CREATE TABLE `t2_1` (`col1` varchar(20) DEFAULT NULL, `col2` double DEFAULT NULL);")
+	tk.MustExec("insert into t1_1 values(12.991);")
+	tk.MustExec("insert into t2_1(col2) values(87), (-9.183), (-9.183);")
+
+	// Analyze to make sure this specific plan can recurring problem.
+	tk.MustExec("analyze table t1_1;")
```

Comment:
Can we use hint instead of analyze?

Reply:
I construct a new case that doesn't rely on specific plan. So no need `analyze` or `hint`.